osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1297 @@
+      break;
+    }
+  }
+}
+
+void getFRUCB(ipmi_entity_t* entity, void* data) {
+  ipmi_fru_t* fru = ipmi_entity_get_fru(entity);
+  if (fru == nullptr) {
+    return;
+  }
+
+  std::unique_ptr<ipmi_fru_t, std::function<void(ipmi_fru_t*)>> freeFru(
+      fru, [](ipmi_fru_t* f) {
+        int rv = ipmi_fru_destroy(f, NULL, NULL);
+        if (rv != 0) {
+          TLOG << "Could not register ipmi_fru_destroy: " << strerror(rv);
+          return;
+        }
+
+      });
+
+  Row r;
+  r["entity_id"] = INTEGER(ipmi_entity_get_entity_id(entity));
+  r["entity_instance"] = INTEGER(ipmi_entity_get_entity_instance(entity));
+
+  char name[IPMI_FRU_NAME_LEN];
+  int len = ipmi_fru_get_name(fru, name, IPMI_FRU_NAME_LEN);
+  if (len < 1) {
+    r["name"] = "unknown";
+  } else {
+    r["name"] = name;
+  }
+
+  ipmi_fru_node_t* node = nullptr;
+
+  const char* type = "";
+  int rv = ipmi_fru_get_root_node(fru, &type, &node);
+  if (rv != 0) {
+    TLOG << "Could not get FRU root node: " << strerror(rv);
+    return;
+  }
+
+  r["type"] = type;
+  traverseFRUNodeTree(node, r);
+
+  IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+  c->insertRowsQueue(
+      r["entity_id"] + "-" + r["entity_instance"] + "-" + r["name"], r);
+}
+
+void iterateMCsCB(ipmi_domain_t* domain, ipmi_mc_t* mc, void* data) {
+  Row r;
+
+  char name[IPMI_MC_NAME_LEN];
+  int len = ipmi_mc_get_name(mc, name, IPMI_MC_NAME_LEN);
+  if (len > 0) {
+    r["name"] = name;
+  }
+
+  r["device_id"] = INTEGER(ipmi_mc_device_id(mc));
+  r["device_revision"] = INTEGER(ipmi_mc_device_revision(mc));
+  r["device_available"] = INTEGER(ipmi_mc_device_available(mc));
+  r["firmware_major_version"] = INTEGER(ipmi_mc_major_fw_revision(mc));
+  r["firmware_minor_version"] = INTEGER(ipmi_mc_minor_fw_revision(mc));
+  r["ipmi_major_version"] = INTEGER(ipmi_mc_major_version(mc));
+  r["ipmi_minor_version"] = INTEGER(ipmi_mc_minor_version(mc));
+  r["iana_manufacturer_id"] = INTEGER(ipmi_mc_manufacturer_id(mc));
+  r["product_id"] = INTEGER(ipmi_mc_product_id(mc));
+  r["provides_device_sdrs"] = INTEGER(ipmi_mc_provides_device_sdrs(mc));
+  r["chassis_support"] = INTEGER(ipmi_mc_chassis_support(mc));
+  r["bridge_support"] = INTEGER(ipmi_mc_bridge_support(mc));
+  r["ipmb_event_generator_support"] =
+      INTEGER(ipmi_mc_ipmb_event_generator_support(mc));
+  r["ipmb_event_reciever_support"] =
+      INTEGER(ipmi_mc_ipmb_event_receiver_support(mc));
+  r["fru_inventory_support"] = INTEGER(ipmi_mc_fru_inventory_support(mc));
+  r["sel_device_support"] = INTEGER(ipmi_mc_sel_device_support(mc));
+  r["sdr_respository_support"] = INTEGER(ipmi_mc_sdr_repository_support(mc));
+  r["sensor_device_support"] = INTEGER(ipmi_mc_sensor_device_support(mc));
+  r["is_active"] = INTEGER(ipmi_mc_is_active(mc));
+
+  unsigned char guid = 0;
+  int rv = ipmi_mc_get_guid(mc, &guid);
+  switch (rv) {
+  case ENOSYS:
+    r["guid"] = "not available";
+    break;
+
+  case 0:
```

Comment:
Above you have `ENOSYS`, is there a preprocessor `#define` for what `0` is here? Could it be swapped out for the macro for consistency?